File: ios/Betty/Features/GroupDetail/GroupGameCard.swift
import SwiftUI

/// Web `Game.vue` (default layout): info row (LIVE badge or date label) above two teams
/// flanking the big score, optional placed-bet chip with awarded points underneath it,
/// 45% dim when finished, optional bet-count chip overlay.
struct GroupGameCard: View {
    let game: Game
    let betted: Bool
    let placedHome: Int
    let placedAway: Int
    let awardedPoints: Int?
    /// True iff the user's own bet on this game is `boosted` AND scored > 0 (spec §2.5
    /// suppression). Set by the caller from `GroupGameCardLogic.awardedBoosted(...)`.
    var awardedBoosted: Bool = false
    /// True iff the user's own bet on this game is `boosted`. Drives the pre-evaluation
    /// rocket next to the placed-bet chip; no points-suppression check (the spec §2.5 rule
    /// only applies to the awarded-points rocket).
    var placedBoosted: Bool = false
    /// nil hides the chip (web `showBets == false`).
    let betCount: Int?
    var onTap: () -> Void

    @Environment(AppEnvironment.self) private var env
    @Environment(ThemeStore.self) private var theme

    var body: some View {
        Button(action: onTap) {
            VStack(spacing: Space.s) {
                infoRow
                teamsRow
            }
            .padding(.top, 14)
            .padding(.horizontal, 16)
            .padding(.bottom, 16)
            .frame(maxWidth: .infinity)
            .background(theme.colors.surface, in: RoundedRectangle(cornerRadius: Radius.sharp))
            .overlay(alignment: .topTrailing) {
                if let betCount {
                    HStack(spacing: 4) {
                        Image(systemName: "person.2")
                            .font(.system(size: 10, weight: .bold))
                        Text("\(betCount)")
                            .font(.bettyKicker)
                            .kerning(0.6)
                    }
                    .foregroundStyle(theme.colors.textSecondary)
                    .padding(.vertical, 4)
                    .padding(.horizontal, 8)
                    .background(theme.colors.overlay08, in: RoundedRectangle(cornerRadius: Radius.sharp))
                    .padding(10)
                }
            }
            .opacity(game.isFinished ? 0.45 : 1)
        }
        .buttonStyle(.plain)
        .accessibilityIdentifier("groupDetail.games.card.\(game.id)")
    }

    private var infoRow: some View {
        HStack {
            switch game.displayState {
            case .live:
                LiveBadge()
            case .fullTime:
                FTBadge()
            case .finished:
                Text("Finished").kicker(theme.colors.textMuted)
            case .scheduled:
                Text(GroupGameDateLabel.text(for: game).uppercased())
                    .kicker(theme.colors.textMuted)
            }
            Spacer()
        }
    }

    private var displayHome: Int? {
        (game.displayState == .live || game.displayState == .fullTime) ? game.liveHomeTeamScore : game.homeTeamScore
    }

    private var displayAway: Int? {
        (game.displayState == .live || game.displayState == .fullTime) ? game.liveAwayTeamScore : game.awayTeamScore
    }

    private var teamsRow: some View {
        HStack(alignment: .top, spacing: Space.xs) {
            teamColumn(env.teamStore.byID(game.homeTeamID))
            VStack(spacing: 6) {
                HStack(alignment: .firstTextBaseline, spacing: 6) {
                    scoreLabel(displayHome)
                    Text("-")
                        .font(.betty(18, .regular))
                        .foregroundStyle(theme.colors.textSecondary)
                    scoreLabel(displayAway)
                }
                if betted {
                    HStack(alignment: .center, spacing: 4) {
                        HStack(alignment: .firstTextBaseline, spacing: 3) {
                            Text("\(placedHome)")
                            Text("-")
                            Text("\(placedAway)")
                        }
                        .font(.bettyKicker)
                        .kerning(0.4)
                        .monospacedDigit()
                        .foregroundStyle(Palette.orange)
                        .padding(.vertical, 3)
                        .padding(.horizontal, 8)
                        .background(Palette.orangeTint15, in: RoundedRectangle(cornerRadius: Radius.sharp))
                        .accessibilityLabel("Your bet \(placedHome) to \(placedAway)")
                        if placedBoosted {
                            Text("🚀")
                                .font(.betty(12, .regular))
                                .accessibilityLabel("Boosted")
                        }
                    }
                }
                if let awardedPoints {
                    HStack(spacing: 4) {
                        Text("\(awardedPoints)P")
                            .kicker(awardedPoints > 0 ? theme.colors.accentPositive : theme.colors.textSecondary)
                        if awardedBoosted && awardedPoints > 0 {
                            // Post-eval rocket (spec §3.4, §2.5 — suppress on 0-point bets).
                            Text("🚀")
                                .font(.betty(12, .regular))
                                .accessibilityLabel("Boosted")
                        }
                    }
                }
            }
            .padding(.top, Space.m)
            teamColumn(env.teamStore.byID(game.awayTeamID))
        }
    }

    private func teamColumn(_ team: Team?) -> some View {
        VStack(spacing: 8) {
            TeamLogoView(team: team, size: 56)
            Text((team?.name ?? "").uppercased())
                .font(.betty(12, .heavy))
                .kerning(0.6)
                .foregroundStyle(theme.colors.textPrimary)
                .lineLimit(1)
                .truncationMode(.tail)
        }
        .frame(maxWidth: .infinity)
    }

    private func scoreLabel(_ score: Int?) -> some View {
        Text(score.map(String.init) ?? "")
            .font(.bettyScore)
            .foregroundStyle(theme.colors.textPrimary)
    }
}

File: android/app/src/main/java/social/betty/features/groupdetail/GroupGameCard.kt
package social.betty.features.groupdetail

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.wrapContentWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.testTag
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import social.betty.core.logic.GameSchedule
import social.betty.core.model.Game
import social.betty.core.model.Team
import social.betty.designsystem.BettyTheme
import social.betty.designsystem.Palette
import social.betty.designsystem.Radius
import social.betty.designsystem.Space
import social.betty.designsystem.components.LiveBadge
import social.betty.designsystem.components.TeamLogo
import java.time.Instant

/**
 * Web `Game.vue` (default layout): info row (LIVE badge or kickoff label) above two teams
 * flanking the big score, optional placed-bet chip with awarded points underneath it,
 * 45% dim when finished, optional bet-count chip overlay.
 */
@Composable
fun GroupGameCard(
    game: Game,
    homeTeam: Team?,
    awayTeam: Team?,
    betted: Boolean,
    placedHome: Int,
    placedAway: Int,
    awardedPoints: Int?,
    /** null hides the chip (web `showBets == false`). */
    betCount: Int?,
    onTap: () -> Unit,
    now: Instant = Instant.now(),
    modifier: Modifier = Modifier,
) {
    val colors = BettyTheme.colors
    val type = BettyTheme.type

    Box(
        modifier = modifier
            .fillMaxWidth()
            .clip(Radius.sharp)
            .background(colors.surface)
            .clickable { onTap() }
            .alpha(if (game.isFinished) 0.45f else 1f)
            .testTag("group-game-card"),
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(top = 14.dp, start = 16.dp, end = 16.dp, bottom = 16.dp),
            verticalArrangement = Arrangement.spacedBy(Space.s),
        ) {
            // Info row
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                if (GameSchedule.isLive(game, now)) {
                    LiveBadge()
                } else {
                    Text(
                        text = GameSchedule.kickoffLabel(game.startDate, now).uppercase(),
                        style = type.kicker,
                        color = colors.textMuted,
                    )
                }
                Spacer(Modifier.weight(1f))
            }

            // Teams + score
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(Space.xs),
                verticalAlignment = Alignment.Top,
            ) {
                teamColumn(homeTeam, Modifier.weight(1f))
                Column(
                    modifier = Modifier.padding(top = Space.m),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(6.dp),
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Text(
                            text = game.homeTeamScore.toString(),
                            style = type.score,
                            color = colors.textPrimary,
                        )
                        Text(
                            text = " - ",
                            style = type.bodyRegular.copy(fontSize = 18.sp),
                            color = colors.textSecondary,
                        )
                        Text(
                            text = game.awayTeamScore.toString(),
                            style = type.score,
                            color = colors.textPrimary,
                        )
                    }
                    if (betted) {
                        Text(
                            text = "$placedHome - $placedAway",
                            style = type.kicker,
                            color = Palette.orange,
                            modifier = Modifier
                                .clip(Radius.sharp)
                                .background(Palette.orangeTint15)
                                .padding(vertical = 3.dp, horizontal = 8.dp),
                        )
                    }
                    if (awardedPoints != null) {
                        Text(
                            text = "${awardedPoints}P",
                            style = type.kicker,
                            color = if (awardedPoints > 0) colors.accentPositive else colors.textSecondary,
                        )
                    }
                }
                teamColumn(awayTeam, Modifier.weight(1f))
            }
        }

        // Bet-count chip (top-trailing).
        if (betCount != null) {
            Text(
                text = "👥 $betCount",
                style = type.kicker,
                color = colors.textSecondary,
                modifier = Modifier
                    .align(Alignment.TopEnd)
                    .padding(10.dp)
                    .clip(Radius.sharp)
                    .background(colors.overlay08)
                    .padding(vertical = 4.dp, horizontal = 8.dp)
                    .testTag("group-game-bet-count"),
            )
        }
    }
}

@Composable
private fun teamColumn(team: Team?, modifier: Modifier) {
    val colors = BettyTheme.colors
    Column(
        modifier = modifier,
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        TeamLogo(url = team?.imageUrl, name = team?.name, size = 56.dp)
        Text(
            text = (team?.name ?: "").uppercase(),
            style = BettyTheme.type.caption.copy(fontWeight = FontWeight(800)),
            color = colors.textPrimary,
            textAlign = TextAlign.Center,
            maxLines = 1,
            overflow = TextOverflow.Ellipsis,
            modifier = Modifier.wrapContentWidth(),
        )
    }
}
